source/connection.c

```diff
@@ -579,6 +587,12 @@ static void s_http_server_clean_up(struct aws_http_server *server) {
     aws_mem_release(server->alloc, server);
 }
 
+static void s_http_server_user_data_clean_up(struct aws_server_user_data *server_user_data) {
```

Comment:
trivial/style: combine this function with the existing s_http_server_clean_up() so there's only 1 clean-up function. 

Cleaner to have 1 clean-up function, than 2 clean-up functions where someone might accidentally call the wrong one